src/include/miopen/median/problem_description.hpp

```diff
@@ -0,0 +1,250 @@
+        if(keepdim)
+        {
+            desired_dims[dim] = 1;
+        }
+        else
+        {
+            desired_dims.erase(desired_dims.begin() + dim);
+            if(desired_dims.size() == 0)
+                desired_dims.push_back(1);
+        }
+
+        if(output_dims != desired_dims || indices_dims != desired_dims)
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MedianForward: Input and output/indices tensor dimension lengths do "
+                         "not match.");
+        }
+
+        return true;
+    }
+
+    bool IsSameType() const
+    {
+        if(inputDesc.GetType() != outputDesc.GetType())
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MedianForward: Input and output tensor data types do not match.");
+        }
+        return true;
+    }
+
+    bool IsValidFloat() const
+    {
+        return inputDesc.GetType() == miopenFloat || inputDesc.GetType() == miopenHalf ||
+               inputDesc.GetType() == miopenBFloat16;
+    }
+
+    NetworkConfig MakeNetworkConfig() const override;
+
+private:
+    const TensorDescriptor& inputDesc;
+    const TensorDescriptor& outputDesc;
+    const TensorDescriptor& indicesDesc;
+    uint64_t dim;
+    bool keepdim;
+};
+
+struct BwdProblemDescription : ProblemDescriptionBase
+{
+    BwdProblemDescription(const TensorDescriptor& outputGradDesc_,
+                          const TensorDescriptor& indicesDesc_,
+                          const TensorDescriptor& inputGradDesc_,
+                          const uint64_t dim_,
+                          const bool keepdim_)
+        : outputGradDesc(outputGradDesc_),
+          indicesDesc(indicesDesc_),
+          inputGradDesc(inputGradDesc_),
+          dim(dim_),
+          keepdim(keepdim_)
+    {
+        IsValidNumDims();
+        IsValidDim();
+        IsRightLength();
+        IsSameType();
+    }
+
+    const TensorDescriptor& GetOutputGradDesc() const { return outputGradDesc; }
+    const TensorDescriptor& GetIndicesDesc() const { return indicesDesc; }
+    const TensorDescriptor& GetInputGradDesc() const { return inputGradDesc; }
+    uint64_t GetDim() const { return dim; }
+    bool GetKeepDim() const { return keepdim; }
+
+    bool IsValidNumDims() const
+    {
+        if(inputGradDesc.GetNumDims() > 5)
+        {
+            MIOPEN_THROW(
+                miopenStatusBadParm,
+                "MedianForward: A tensor with dimensions greater than 5 is not supported yet.");
+        }
+
+        return true;
+    }
+
+    bool IsValidDim() const
+    {
+
+        if(dim >= inputGradDesc.GetNumDims())
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "MedianForward: Invalid dim " + std::to_string(dim) +
+                             " for input tensor with " +
+                             std::to_string(inputGradDesc.GetNumDims()) + " dimensions");
+        }
+
+        return true;
+    }
```

Comment:
The same as above. Support negative `dim` value.

Reply:
pls check my [cmt](https://github.com/moreh-dev/MIOpen/pull/71#discussion_r1895528156) above

Reply:
I understand your intention.

However, regarding the reference you provided—specifically the `cumsum` operation—the `dim` parameter is required, making it reasonable to support negative dimensions in the backend layer. For operations where `dim` is optional, how would you handle cases where it’s left unspecified? There are, of course, multiple approaches. In my opinion, the way modnn handles this—defaulting to `-1` and supporting only positive `dim`—is quite reasonable.

Therefore, unless the MIOpen convention explicitly requires handling negative dimensions in the backend layer, I suggest making it an optional choice, and I suggest let's keep this code section as it is for now (because many previous MIOpen ops already use unsigned types for param `dim` [[ref]](https://github.com/ROCm/MIOpen/blob/31be57bb7c2135a845ab6201dde54ec28c0b5c28/driver/layernorm_driver.hpp#L200-L201).

Moreover, one of the uses of the dim parameter in MIOpen looks like this:
```
auto reduce_size = inputDesc.GetLengths()[dim];
```

As shown, dim is used as an index in a vector. I understand that handling negative dim values is managed elsewhere before this point. However, wouldn't it be safer to explicitly prevent `dim` from being negative at this backend level and delegate the responsibility of handling negative dim values to a higher-level abstraction?